src/app/use-cases/request-signature.ts

```diff
@@ -71,3 +103,51 @@ export const makeRequestSignature =
       ),
       TE.chainFirst(addSignatureRequest)
     );
+
+/*
+ * Update status of signature request only if:
+ * new status is READY
+ * and previous status is WAIT_FOR_ISSUER!
+ */
+export const updateStatusRequestSignature =
+  (
+    upsertSignatureRequest: UpsertSignatureRequest,
+    getSignatureRequest: GetSignatureRequest
+  ) =>
+  (
+    payload: RequestSignatureStatusPayload
+  ): TE.TaskEither<Error, SignatureRequest> =>
+    pipe(
+      payload.signatureRequestStatus,
+      validate(readyStatus, "Only READY status is allowed!"),
+      TE.fromEither,
+      TE.chain((signatureRequestStatus) =>
+        pipe(
+          getSignatureRequest(
+            payload.signatureRequestId,
+            payload.subscriptionId
+          ),
+          TE.chainEitherKW(
+            flow(
+              E.fromOption(
+                () =>
+                  new EntityNotFoundError("Error getting the Signature Request")
+              ),
+              E.filterOrElse(
+                (request) => request.status === "WAIT_FOR_ISSUER",
+                constant(
+                  new InvalidEntityError(
+                    "The status can only be changed if the signature request is still in WAIT_FOR_ISSUER!"
```

Comment:
try to isolate this logic into `nextStatus`

Reply:
`nextStatus` is meant to calculate a new state from a previous one, in this case we just need to check if the signature request is in the" WAIT_FOR_ISSUER "state. 

However do you think it is more logical to create such a function like:
`const nextStatus: (request: SignatureRequest) => (newStatus: SignatureRequestStatus) => O.Option<SignatureRequestStatus>` which checks if the new state is admitted and returns it?

Reply:
ok.

probably, you choose.